spock-specs/src/test/groovy/org/spockframework/smoke/StackTraceFiltering.groovy

```diff
@@ -309,6 +313,32 @@ apackage.ASpec|a feature|2
     """
   }
 
+  def 'stack traces are filtered on rendering expression info value'() {
+    given:
+    def expressionInfo = new ExpressionInfo(null, null, null, emptyList())
+    expressionInfo.value = new Exception()
+    expressionInfo.value.stackTrace = [
+      new StackTraceElement('org.codehaus.groovy.runtime.Foo', 'bar', null, 0),
+      new StackTraceElement('org.codehaus.groovy.reflection.Foo', 'bar', null, 0),
+      new StackTraceElement('org.codehaus.groovy.FooMetaClassBaz', 'bar', null, 0),
+      new StackTraceElement('groovy.FooMetaClassBaz', 'bar', null, 0),
+      new StackTraceElement('groovy.lang.MetaMethod', 'bar', null, 0),
+      new StackTraceElement('java.lang.reflect.Foo', 'bar', null, 0),
+      new StackTraceElement('sun.reflect.Foo', 'bar', null, 0),
+      new StackTraceElement('jdk.internal.reflect.Foo', 'bar', null, 0),
+      new StackTraceElement('org.spockframework.runtime.Foo', 'bar', null, 0),
+      new StackTraceElement('org.spockframework.runtime.foo.Baz', 'bar', null, 0)
```

Comment:
A nice way of stackTrace creation :).

It reminded me that I wanted to take a closer look at `StackWalker` in Java 9 :).